drop-transfer/src/file/mod.rs

```diff
@@ -296,19 +297,35 @@ impl FileToSend {
     }
 
     /// Calculate sha2 of a file. This is a blocking operation
-    pub(crate) async fn checksum(&self, limit: u64) -> crate::Result<[u8; 32]> {
+    pub(crate) async fn checksum<F, Fut>(
+        &self,
+        limit: u64,
+        progress_cb: Option<F>,
```

Comment:
Why option? You could pass an empty function if you want to ignore the progress report